Antidote/AbstractCallViewController.h

```diff
@@ -18,21 +16,20 @@
 + (instancetype)new NS_UNAVAILABLE;
 
 /**
- * Create an instance of the CallViewController.
- * @param chat Appropriate chat for the call.
- * @param manager The call maanger for the application.
+ * Create an instance of the AbstractCallViewController.
+ * @param nickname The nickname associated with the caller.
  */
-- (instancetype)initWithCall:(OCTCall *)call submanagerCalls:(OCTSubmanagerCalls *)manager;
+- (instancetype)initWithCallerNickname:(NSString *)nickname;
 
 /**
- * The call associated with this view controller.
+ * Abstract delegate
  */
-@property (strong, nonatomic, readonly) OCTCall *call;
+@property (weak, nonatomic) id delegate;
```

Comment:
It wouldn't be less convenient as compiler won't check if CallsManager confirms to specific protocol. Basically it is better to avoid dynamic behavior when possible, compiler is your friend.

I think that following will be fine
```
switch (call.status) {
    case OCTCallStatusActive: {
        ActiveCallViewController *activeVC = [ActiveCallViewController new];
        activeVC.delegate = self;
        viewController = activeVC;
        break;
    }
    case OCTCallStatusDialing: {
        DialingCallViewController *dialingVC = [DialingCallViewController new];
        dialingVC.delegate = self;
        viewController = dialingVC;
        break;
    }
    case OCTCallStatusRinging: {
        RingingCallViewController *ringingVC = [RingingCallViewController new];
        ringingVC.delegate = self;
        viewController = ringingVC;
        break;
    }
    case OCTCallStatusPaused:
        NSAssert(NO, @"We should not be here yet. Not yet implemented");
        break;
}
```